src/libsyntax/ext/expand.rs

```diff
@@ -229,10 +226,31 @@ fn expand_mac_invoc<T>(mac: ast::Mac, ident: Option<Ident>, attrs: Vec<ast::Attr
                     export: attr::contains_name(&attrs, "macro_export"),
                     allow_internal_unstable: attr::contains_name(&attrs, "allow_internal_unstable"),
                     attrs: attrs,
-                });
+                };
 
-                // macro_rules! has a side effect but expands to nothing.
-                Some(Box::new(MacroScopePlaceholder))
+                // DON'T mark before expansion.
```

Comment:
Pre-existing, but this comment should have been removed in #34570 and is now outdated.